docs/debugging.md

```diff
@@ -0,0 +1,325 @@
+
+![Screenshot showing the output of graphql_debug()](./debugging-output-graphql-debug.png)
+
+Various breakpoints can be added throughout resolution and tagged with a Type or other meta to help filter the logs and identify how things are resolving.
+
+This can add a lot of value to developer workflows.
+
+## Internal Server Error
+
+If you see the error "Internal Server Error" this means there was an error that *might* not be safe to show details of to public users. It might expose some implementation details, and so it is masked as "Internal Server Error". You can [enable debug mode](#enabling-debug-mode) to show more detailed errors. 
+
+One common reason for seeing this error is when a field or Type is registered with missing information. 
+
+Take the following example:
+
+**Invalid Field (missing **type**)**
+
+Registering a field without a type would throw an `Internal Server Error` 
+
+```php
+add_action( 'graphql_register_types', function() {
+  register_graphql_field( 'Post', 'invalidField', [
+    'description' => __( 'Invalid field with no Type', 'your-textdomain' ),
+  ] );
+});
+```
+
+**Valid Field** **(includes type)**
+
+Adding the type would cause the error to go away.
+
+```php
+add_action( 'graphql_register_types', function() {
+  register_graphql_field( 'Post', 'validField', [
+    'type' => 'String',
+    'description' => __( 'Valid field with a Type defined', 'your-textdomain' ),
+  ] );
+});
+```
+
+## Tracing
+
+WPGraphQL allows you to enable GraphQL Tracing. Tracing tracks how long each resolver in a GraphQL request takes to complete and adds the data to the results of the request as an extension. This can be helpful for debugging problematic resolvers that might be slowing down your application.
+
+### Enable Tracing
+
+From the WPGraphQL Settings page, check the box labeled "Enable GraphQL Tracing".
+
+![Screenshot of the WPGraphQL Setting to Enable GraphQL Tracing](./debugging-setting-graphql-enable-tracing.png)
+
+### Trace Data
+
+Next time you execute a request, you will see Trace data in the response, like so: 
+
+![Screenshot showing GraphQL Trace log](./debugging-graphql-trace-log.png)
+
+The Trace logs include data for the entire request:
+
+- startTime
+- endTime
+- duration
+
+As well as data for each field within the request:
+
+- path
+- parentType
+- fieldName
+- returnType
+- startOffset
+- duration
+
+This data can be very helpful when debugging bottlenecks.
+
+Let's take a look at an example.
+
+Executing the following query:
+
+```graphql
+{
+  posts {
+    nodes {
+      id
+      title
+    }
+  }
+}
+```
+
+We would see the following Trace data for the request:
+
+```json
+  ...,
+  "tracing": {
+    "version": 1,
+    "startTime": "1600294443.335",
+    "endTime": "1600294443.381",
+    "duration": 45998,
+    "execution": {
+      ...
```

Comment:
Github is syntax highlighting this like an error. 

Want to see if there actually is some issue or if Github is just being a turd.

Reply:
![Screen Shot 2021-05-04 at 2 26 10 PM](https://user-images.githubusercontent.com/1260765/117065381-ad99f080-ace4-11eb-82d4-fb5bafa2dcef.png)